jax/lib/xla_bridge.py

```diff
@@ -110,61 +113,80 @@ def get_compile_options(num_replicas, num_partitions, device_assignment=None):
 
   return compile_options
 
-_backends = {}
+_backend_factories = {}
 
-def register_backend(name, factory):
-  _backends[name] = factory
+def register_backend(name: str, factory: Callable[[Optional[str]], LocalClient]):
+  _backend_factories[name] = factory
 
-def _get_local_backend(platform=None):
-  if not platform:
-    platform = FLAGS.jax_platform_name or None
 
+# TODO(skye): remove this function once xla_client.get_local_backend() never
+# returns None.
+def _get_local_backend(platform):
   backend = xla_client.get_local_backend(platform)
   if backend is None:
     raise RuntimeError("No local XLA backends found.")
```

Comment:
This message string seems misleading given you call this function from multiple places.